### PR TITLE
Remove fail-on-empty option (and fix default behavior)

### PR DESCRIPTION
**Pull request by chadwhitacre**
The default for `fail-on-empty` is [not working properly](https://github.com/getsentry/snuba/actions/runs/4982169145/jobs/8917438053?pr=4174). Nobody is asking for the option, we should just drop back to fixing the default behavior without adding the option.

### CODE REVIEW

**Comment by chadwhitacre** [comment]
I believe [this demonstrates](https://github.com/getsentry/action-github-commit/actions/runs/4995165678/jobs/8946696896#step:10:43) the misbehavior we saw in `snuba`.

File: dist/index.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ function run() {
             const { owner, repo } = github.context.repo;
             const token = core.getInput('github-token');
             const message = core.getInput('message') || 'Default commit message';
-            const failOnEmpty = core.getInput('fail-on-empty') || 'false';
+            const failOnEmpty = (core.getInput('fail-on-empty') || 'false') === 'true';
```
Comment (chadwhitacre) [inline dist/index.js:55]: This is the bugfix.

File: src/main.ts [hunk 2]
```diff
@@ -9,7 +9,7 @@ async function run(): Promise<void> {
     const {owner, repo} = github.context.repo;
     const token = core.getInput('github-token');
     const message = core.getInput('message') || 'Default commit message';
-    const failOnEmpty = core.getInput('fail-on-empty') || 'false';
+    const failOnEmpty = (core.getInput('fail-on-empty') || 'false') === 'true';
```
Comment (chadwhitacre) [inline src/main.ts]: This is the bugfix.

**Review by asottile-sentry — COMMENTED** [review]
I don't think `fail-on-empty` should be an option of this action at all.  it should be up to the calling code to decide whether that's an error

the previous behaviour was this action was a noop if there were no changes -- we should preserve that (without an option that nobody will use, ideally)

adding this "feature" also makes it more difficult to switch to a maintained action instead of continuing to build features into ours

**Comment by chadwhitacre** [comment]
> I don't think `fail-on-empty` should be an option

Done in 9805ada4e9c70789c06993c6e90472c7305174fd.

**Comment by hubertdeng123** [comment]
Is there an example of where it's used? Looks like snuba is using the default value for `fail-on-empty` and a quick search comes up with nothing

https://github.com/search?q=org%3Agetsentry+fail-on-empty&type=code

**Comment by chadwhitacre** [comment]
> Done in 9805ada4e9c70789c06993c6e90472c7305174fd.

... and 24fb9f264aad66e860b38744ca0dd152e9f3f5c6. 😬

**Comment by chadwhitacre** [comment]
> Is there an example of where it's used?

No, I only added it in #17 a few days ago, nobody is using it. As @asottile-sentry points out the important thing is preserving the existing default behavior (no-op when there are no changes). The option is YAGNI.

**Review by hubertdeng123 — APPROVED** [review]

**Pull request merged**